netbox_cmdb/netbox_cmdb/views.py

```diff
@@ -48,6 +63,110 @@
 )
 
 
+## Decommission a device
+class DecommissioningView(ObjectDeleteView):
+    queryset = Device.objects.all()
+    template_name = "netbox_cmdb/decommissioning.html"
+
+    def get(self, request, *args, **kwargs):
+        """
+        GET request handler.
+
+        Args:
+            request: The current request
+        """
+        obj = self.get_object(**kwargs)
+        form = ConfirmationForm(initial=request.GET)
+
+        # If this is an HTMX request, return only the rendered deletion form as modal content
+        if is_htmx(request):
+            # form_url = reverse("decommisioning_delete", kwargs={'pk': obj.pk})
+            form_url = f"/plugins/cmdb/decommisioning/{kwargs['pk']}/delete"
+            return render(
+                request,
+                "htmx/delete_form.html",
+                {
+                    "object": obj,
+                    "object_type": self.queryset.model._meta.verbose_name,
+                    "form": form,
+                    "form_url": form_url,
+                    **self.get_extra_context(request, obj),
+                },
+            )
+
+        return render(
+            request,
+            self.template_name,
+            {
+                "object": obj,
+                "form": form,
+                "return_url": self.get_return_url(request, obj),
+                **self.get_extra_context(request, obj),
+            },
+        )
+
+    def post(self, request, *args, **kwargs):
+        # Fetch the device to delete
+        device = self.get_object(**kwargs)
+        deleted_objects = {
+            "bgp_sessions": [],
+            "device_bgp_sessions": [],
+            "bgp_peer_groups": [],
+            "route_policies": [],
+            "prefix_lists": [],
+            "bgp_community_lists": [],
+            "snmp": [],
+        }
+
+        device_name = device.name
+
+        try:
+            with transaction.atomic():
```

Comment:
we could put this common logic in a same function since this is used by the API endpoint and the UI view, wdyt ?

Reply:
WIP for another review, as I am working on adding the decommissioning of a site.

It was my initial goal. BUT, I don't think we can really factorize the UI and the API calls.
The API is using obj__name while the UI uses obj__id.

Reply:
but spoiler, we can only factorize if we get the object by ID and not by name on the API side.
This change behavior in case of device duplicates.